lib/model/AjvValidator.js

```diff
@@ -157,25 +157,19 @@ function parseValidationError(errors, modelClass, options) {
 
   for (let i = 0; i < errors.length; ++i) {
     const error = errors[i];
-    let key = error.dataPath.substring(1);
-
-    const params = error.params;
-    if (!key && params) {
-      key = params.missingProperty || params.additionalProperty;
-    }
-
-    if (options.dataPath) {
-      key = `${options.dataPath.substring(1)}.${key}`;
-    }
-
+    const dataPath = `${options.dataPath || ''}${error.dataPath}`;
+    // Unknown properties are reported in `['propertyName']` notation,
+    // so replace those with dot-notation, see:
+    // https://github.com/epoberezkin/ajv/issues/671
+    const key = dataPath.replace(/\['([^']*)'\]/g, '.$1').substring(1);
```

Comment:
Doesn't this replace all brackets with dot notation? For example `foo['words lol'].bar` --> `foo.words lol.bar` which is an invalid path?

Reply:
Note that the dataPaths all start with `.`and we remove it with `.substring(1)`

Reply:
The correct path for `foo['words lol'].bar` is `foo['words lol'].bar`, not `foo.words.lol.bar`.

Reply:
Keys can have spaces and in that case bracket notation must be used.

Reply:
Actually it seems that lodash accepts spaces with dot notation:
```js
_.get({a: {"b c": 1}}, "a.b c"); // --> 1
```

Reply:
Oh I sorry, I didn't see the space because of the way the line broke.

Reply:
that can be easily remedied. New version coming up.

Reply:
This works:
```js
`foo['words lol'].bar['additional']`.replace(/\['([^' ]*)'\]/g, '.$1')
```

Reply:
Which version do you prefer?

Reply:
Also, are you sure that Objection relations or properties can have spaces in them?

Reply:
The only difference is the added space in `[^']` → `[^' ]`. So either scenario can be addressed easily. I've pushed the version that handles the space in e8ec684982e2b16b6ae7319a9130eff82a2ab30f.